projects/igniteui-angular-elements/src/app/custom-strategy.ts

```diff
@@ -284,6 +314,11 @@ class IgxCustomNgElementStrategy extends ComponentNgElementStrategy {
         return children;
     }
 
+    private addToParentCache(parentElement: IgcNgElement, queryName: string) {
```

Comment:
I suppose the cache being on the strategy will go with the root HGrid element and normally in Angular's runtime those child embedded views never get destroyed and sort of do the same? Still not a bad idea to check and perhaps if those leak and should be removed on destroy. 
Didn't notice a heap buildup but did one for DOM nodes, which might be sitting retained and could be of concern